src/fields.js

```diff
@@ -2544,88 +3347,296 @@ x3dom.fields.Quaternion.rotateFromTo = function ( fromVec, toVec )
     return new x3dom.fields.Quaternion( axis.x, axis.y, axis.z, s );
 };
 
+/**
+ * Returns a four-element array of this quaternion's rotation in
+ * an axis-angle representation suitable for communication with OpenGL.
+ *
+ * @returns {Number[]} an array of four numbers holding the rotation
+ *                     axis' x-, y-, and z-coordinates, as well as the
+ *                     angle in radians
+ */
 x3dom.fields.Quaternion.prototype.toGL = function ()
 {
     var val = this.toAxisAngle();
     return [ val[ 0 ].x, val[ 0 ].y, val[ 0 ].z, val[ 1 ] ];
 };
 
+/**
+ * Returns a string representation of this quaternion.
+ *
+ * @returns {String} a string representation of this quaternion
+ */
 x3dom.fields.Quaternion.prototype.toString = function ()
 {
     return this.x + " " + this.y + " " + this.z + ", " + this.w;
 };
 
+/**
+ * Parses the axis-angle representation of a rotation from a string
+ * and sets this quaternion's rotation from it, finally returning this
+ * quaternion itself.
+ *
+ * @param {String} str - the string to parse the axis-angle data from
+ * @returns {x3dom.fields.Quaternion} this modified quaternion
+ */
 x3dom.fields.Quaternion.prototype.setValueByStr = function ( str )
 {
     var m = /^\s*([+\-]?\d*\.*\d*[eE]?[+\-]?\d*?)\s*,?\s*([+\-]?\d*\.*\d*[eE]?[+\-]?\d*?)\s*,?\s*([+\-]?\d*\.*\d*[eE]?[+\-]?\d*?)\s*,?\s*([+\-]?\d*\.*\d*[eE]?[+\-]?\d*?)\s*$/.exec( str );
     m = m || [ 0, 1, 0, 0, 0 ];
-    var quat = x3dom.fields.Quaternion.axisAngle( new x3dom.fields.SFVec3f( +m[ 1 ], +m[ 2 ], +m[ 3 ] ), +m[ 4 ] );
+    var quat = x3dom.fields.Quaternion.axisAngle(
+        new x3dom.fields.SFVec3f( +m[ 1 ], +m[ 2 ], +m[ 3 ] ), +m[ 4 ]
+    );
     this.x = quat.x;
     this.y = quat.y;
     this.z = quat.z;
     this.w = quat.w;
     return this;
 };
 
+// Definition of CSS color names
+var colorNames = {
```

Comment:
Makes `colorNames` global. Should stay inside scope of `_colorParse`. Not sure why `_colorParse` itself is global.